pkg/controller/perconaservermongodb/volumes.go

```diff
@@ -124,35 +192,26 @@ func (r *ReconcilePerconaServerMongoDB) resizeVolumesIfNeeded(ctx context.Contex
 			}
 		}
 
-		if !resizeInProgress {
-			if err := k8s.DeannotateObject(ctx, r.client, cr, psmdbv1.AnnotationPVCResizeInProgress); err != nil {
-				return errors.Wrap(err, "deannotate psmdb")
+		if updatedPVCs == len(pvcsToUpdate) {
+			log.Info("Deleting statefulset")
+
+			if err := r.client.Delete(ctx, sts, client.PropagationPolicy("Orphan")); err != nil {
+				if k8serrors.IsNotFound(err) {
+					return nil
+				}
+				return errors.Wrapf(err, "delete statefulset/%s", sts.Name)
 			}
 
-			log.Info("PVC resize completed")
+			if err := k8s.DeannotateObject(ctx, r.client, sts, psmdbv1.AnnotationPVCResizeInProgress); err != nil {
+				return errors.Wrap(err, "deannotate pxc")
+			}
```

Comment:
Why should we deannotate a statefulset if it has already been deleted?

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/1588/commits/bc5963eee2d68908711a928f40ff2e10a99ef48c